extracted/vm/src/win/aioWin.c

```diff
@@ -506,16 +497,6 @@ EXPORT(long) aioPoll(long microSeconds){
 		CloseHandle(waitingHandles[i]);
 	}
 
-
```

Comment:
Which is the reason to not having the quick return when the timeout has arrived? Because we are testing the handlers when we have already testing them

Reply:
Hi Pablo,

Because not all the threads may have had a chance to execute before the timeout arrives.  If the quick return is used there is a good chance that sockets will incorrectly not be signaled as having IO complete - we saw this happen quite regularly.

So the `WaitForMultipleObjectsEx()` call will return early if there is an event, and may return early if there is IO, allowing the event or IO to be processed, but if it times out it isn't really giving any information about IO completion.  And checking whether IO has occurred is very quick, 1000 sockets are checked in less than a millisecond (from memory).

Reply:
Ok, perfect, thanks for the explanation